table/get_context.cc

```diff
@@ -88,10 +89,33 @@ void GetContext::SaveValue(const Slice& value, SequenceNumber seq) {
 }
 
 void GetContext::RecordCounters(Tickers ticker, size_t val) {
-  if (ticker == Tickers::TICKER_ENUM_MAX) {
-    return;
+  for (auto it = ticker_pairs_.begin(); it != ticker_pairs_.end(); ++it) {
+    if (it->first == ticker) {
+      it->second += static_cast<uint64_t>(val);
+      return;
+    }
   }
-  tickers_value[ticker] += static_cast<uint64_t>(val);
+  ticker_pairs_.push_back(std::make_pair(ticker, static_cast<uint64_t>(val)));
+}
+
+void GetContext::InitTickers() {
+  ticker_pairs_.push_back(std::make_pair<Tickers, uint64_t>(BLOCK_CACHE_HIT, 0));
+  ticker_pairs_.push_back(std::make_pair<Tickers, uint64_t>(BLOCK_CACHE_INDEX_HIT, 0));
+  ticker_pairs_.push_back(std::make_pair<Tickers, uint64_t>(BLOCK_CACHE_DATA_HIT, 0));
+  ticker_pairs_.push_back(std::make_pair<Tickers, uint64_t>(BLOCK_CACHE_FILTER_HIT, 0));
+  ticker_pairs_.push_back(std::make_pair<Tickers, uint64_t>(BLOCK_CACHE_INDEX_MISS, 0));
+  ticker_pairs_.push_back(std::make_pair<Tickers, uint64_t>(BLOCK_CACHE_FILTER_MISS, 0));
+  ticker_pairs_.push_back(std::make_pair<Tickers, uint64_t>(BLOCK_CACHE_DATA_MISS, 0));
+  ticker_pairs_.push_back(std::make_pair<Tickers, uint64_t>(BLOCK_CACHE_BYTES_READ, 0));
+  ticker_pairs_.push_back(std::make_pair<Tickers, uint64_t>(BLOCK_CACHE_MISS, 0));
+  ticker_pairs_.push_back(std::make_pair<Tickers, uint64_t>(BLOCK_CACHE_ADD, 0));
+  ticker_pairs_.push_back(std::make_pair<Tickers, uint64_t>(BLOCK_CACHE_BYTES_WRITE, 0));
+  ticker_pairs_.push_back(std::make_pair<Tickers, uint64_t>(BLOCK_CACHE_INDEX_ADD, 0));
+  ticker_pairs_.push_back(std::make_pair<Tickers, uint64_t>(BLOCK_CACHE_INDEX_BYTES_INSERT, 0));
+  ticker_pairs_.push_back(std::make_pair<Tickers, uint64_t>(BLOCK_CACHE_DATA_ADD, 0));
+  ticker_pairs_.push_back(std::make_pair<Tickers, uint64_t>(BLOCK_CACHE_DATA_BYTES_INSERT, 0));
+  ticker_pairs_.push_back(std::make_pair<Tickers, uint64_t>(BLOCK_CACHE_FILTER_ADD, 0));
+  ticker_pairs_.push_back(std::make_pair<Tickers, uint64_t>(BLOCK_CACHE_FILTER_BYTES_INSERT, 0));
```

Comment:
Wow, I didn't realize the list is that long. How about we make a GetStats object and make them all variables? We already have things like this: https://github.com/facebook/rocksdb/blob/5.11.fb/db/compaction_iteration_stats.h and https://github.com/facebook/rocksdb/blob/5.11.fb/db/internal_stats.h#L133-L161 which don't feel very hard to maintain.